packages/tpetra/core/src/Tpetra_Import_Util2.hpp

```diff
@@ -318,9 +359,11 @@ sortAndMergeCrsEntries (const Teuchos::ArrayView<size_t> &CRS_rowptr,
     CRS_rowptr[i] = old_curr;
     if(old_rowptr_i >= nnz) continue;
 
-    Scalar* locValues   = &CRS_vals[old_rowptr_i];
     size_t NumEntries   = CRS_rowptr[i+1] - old_rowptr_i;
-    Ordinal* locIndices = &CRS_colind[old_rowptr_i];
+    Teuchos::ArrayRCP<Scalar> locValues;
```

Comment:
A bit of history: @csiefer2 had evidence a while back that compilers weren't so good at optimizing through `Teuchos::ArrayRCP::operator[]`.  This is why the code uses raw pointers instead of `Teuchos::ArrayRCP`.  It's not clear how much of that is ArrayRCP's issue and how much is just the code being too long for the compiler to optimize effectively, but it might still be of concern.  On the other hand, `Teuchos::ArrayRCP` (and `Kokkos::View`) can do bounds checking, and may thus save us future trouble.